feat(contracts): remove validation reward penalty (#137)
kroma-network/kroma: specs/validations.md

<!-- All glossary references in this file. -->

[g-l1]: glosarry.md#l1

> 

Reviewer: Please remove these empty lines.

- [Submitting L2 Output Commitments](#submitting-l2-output-commitments)
- [L2 Output Commitment Construction](#l2-output-commitment-construction)

>     - [Output Payload(Version 0)](#output-payloadversion-0)

Reviewer: Please remove these redundant indents.

A validator who submits an output can receive a reward from L2 `ValidatorRewardVault` contract when the output is
finalized, it is called validation reward. When the output is finalized, the `ValidatorPool` contract sends a message
to pay the reward in the L2 `ValidatorRewardVault` via the `KromaPortal` contract.

> The reward is the balance in the `ValidatorRewardVault` when the output is finalized, divided by `REWARD_DIVIDER`,
> which is a value equal to the number of outputs in a week.

Reviewer 2: ```suggestion
The reward is calculated by `ValidatorRewardVault` divided by `REWARD_DIVIDER`,
which is the number of outputs in a week.
```

A validator who submits an output can receive a reward from L2 `ValidatorRewardVault` contract when the output is
finalized, it is called validation reward. When the output is finalized, the `ValidatorPool` contract sends a message
to pay the reward in the L2 `ValidatorRewardVault` via the `KromaPortal` contract.

The `ROUND_DURATION` time is divided into `NON_PENALTY_PERIOD` and `PENALTY_PERIOD`.
The `NON_PENALTY_PERIOD` is the time to guarantee that transaction will be included in the L1 block.
If the validator submits the output within this time, it can receive the full reward.
The full reward is the balance in the `ValidatorRewardVault` when the output is finalized, divided by `REWARD_DIVIDER`,
which is a value equal to the number of outputs in a week. If the validator submits an output after the
`NON_PENALTY_PERIOD`, the reward will gradually decrease, it is called a penalty. The percentage of the penalty is

> calculated using the time elapsed during the `PENALTY_PERIOD`, and the reward is reduced by this percentage.

Reviewer: I think below lines are necessary.
```
The reward is the balance in the `ValidatorRewardVault` when the output is finalized, divided by `REWARD_DIVIDER`,
which is a value equal to the number of outputs in a week.
```

Rewards received by the validator can be withdrawn to L1 via `withdraw()` in the ValidatorRewardVault.

### Configuration of ValidatorPool

The `NON_PENALTY_PERIOD` is the period during a submission round that is not penalized (in seconds).
The `PENALTY_PERIOD` is the period during a submission round that is penalized (in seconds).

The sum of the two values, `NON_PENALTY_PERIOD` and `PENALTY_PERIOD`, must be equal to `ROUND_DURATION`.

> `ROUND_DURATION` is equal to `(L2_BLOCK_TIME * SUBMISSION_INTERVAL) / 2`

Reviewer: I think below lines are necessary.
```
### Configuration of ValidatorPool

`ROUND_DURATION` is equal to `(L2_BLOCK_TIME * SUBMISSION_INTERVAL) / 2`.
```
